src/App.jsx

```diff
@@ -14,33 +15,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALF = 'alf';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getSortInfo(goods, { sortField, isRevesed }) {
+  let prepedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    prepedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SORT_FIELD_ALF:
+          return a.localeCompare(b);
+        case SORT_FIELD_LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isRevesed) {
+    prepedGoods = prepedGoods.reverse();
+  }
+
+  return prepedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isRevesed, setIsRevesed] = useState(false);
+
+  const start = getSortInfo(goodsFromServer, {
```

Comment:
start is not good naming.
Suggest `visiibleItems` for example for such constant.